llms/mlx_lm/models/phi2.py

```diff
@@ -10,43 +9,87 @@
 
 @dataclass
 class ModelArgs(BaseModelArgs):
-    n_positions: int = 2048
+    max_position_embeddings: int = 2048
     vocab_size: int = 51200
-    n_embd: int = 2560
-    n_head: int = 32
-    n_layer: int = 32
-    rotary_dim: int = 32
+    hidden_size: int = 2560
+    num_attention_heads: int = 32
+    num_hidden_layers: int = 32
+    num_key_value_heads: int = 32
+    partial_rotary_factor: float = 0.4
+    intermediate_size: int = 10240
+    layer_norm_eps: float = 1e-5
+    rope_theta: float = 10000.0
+
+    def __post_init__(self):
+        if self.num_key_value_heads is None:
+            self.num_key_value_heads = self.num_attention_heads
 
 
 class LayerNorm(nn.LayerNorm):
```

Comment:
I have tried to replace the LayerNorm with the native `nn.LayerNorm`, but for some reason, not converting to fp32 during the forward pass led to a noticeable decline in inference.

Reply:
Yes for any operation which does a sum (or reduction over many elements, fp16 will typically not work well). We cast up to fp32 in RMSNorm and LayerNorm (as well as softmax for the denominator).

It might be nice to make this an option for our normalization layers so we can avoid so much code duplication..

Reply:
That makes sense. Thank you for the detailed explanation.